modules/custom/az_core/az_core.module

```diff
@@ -100,3 +100,64 @@ function az_core_block_access(Block $block, $operation, AccountInterface $accoun
   }
   return AccessResult::neutral();
 }
+
+
+/**
+ * Implements hook_token_info().
+ */
+
+function az_core_token_info() {
+  return [
+    'tokens' => [
+      'node' => [
+        'az-canonical-url' => [
+          'name' => t("Canonical URL"),
+          'description' => t("Switches between field_az_link and node:url."),
+        ],
+      ],
+    ],
+  ];
+}
+
+/**
+ * Implements hook_token().
+ */
+
+function az_core_tokens($type, $tokens, array $data, array $options, \Drupal\Core\Render\BubbleableMetadata $bubbleable_metadata) {
+  $replacements = [];
+  if ($type == 'node') {
+
+    foreach ($tokens as $name => $original) {
+      // Find the desired token by name.
+      switch ($name) {
+        case 'az-canonical-url':
+          $replacements[$original] = az_core_canonical_url($data);
+          break;
+      }
+    }
+  }
+  return $replacements;
+
+}
+/**
+ * @TODO internal paths don't work.
+ */
+function az_core_canonical_url($data) {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $data['node'];
+  $options = ['absolute' => TRUE];
+  $url = \Drupal\Core\Url::fromRoute('entity.node.canonical', ['node' => $node->id()], $options);
+  $default_url = $url->toString();
```

Comment:
Since the field formatter provided by Field Group Link is [actually expecting a URI](https://git.drupalcode.org/project/field_group_link/-/blob/8.x-3.x/src/Plugin/field_group/FieldGroupFormatter/Link.php#L205-211) that it will generate a URL from, I think we might want to do something like this here:
```suggestion
  $default_url = $url->getUri();
```